core/src/main/java/com/linecorp/armeria/client/circuitbreaker/NonBlockingCircuitBreaker.java

```diff
@@ -120,6 +120,12 @@ private boolean checkIfExceedingFailureThreshold(EventCount count) {
 
     @Override
     public boolean canRequest() {
+        final State currentState = state.get();
+        return currentState.isClosed() || currentState.isHalfOpen() && currentState.checkTimeout();
```

Comment:
This could be a huge breaking change for someone who uses `CircuitBreaker` standalone.
How about deprecating `canRequest()` and adding the following new methods?
```
boolean available();
boolean acquire();
```

Reply:
How about:

```java
CircuitState state();
boolean tryRequest();
```

Reply:
> CircuitState state();

I had a similar idea. But `canRequest()` also see `State.checkTimeout()`. Is it better to expose `State` as `CircuitBreakerState` after refining the `State`?

Reply:
SGTM

Reply:
Thanks for comments! Initially, I already thought returning `State` as well. But I concerned that client user should know the inner class. But LGTM if we can decide it to upper level. 

Updated PR. PTAL 🙏